asdf/tags/core/integer.py

```diff
@@ -16,87 +15,38 @@ class IntegerType(_types.AsdfType):
     When reading files that contain arbitrarily large integers, the values that
     are restored in the tree will be raw Python `int` instances.
 
+    DEPRECATED.  Large integer values are now handled automatically and need
+    no special wrapper class.  Use asdf.get_config().array_inline_threshold
+    to customize the array storage type.
+
     Parameters
     ----------
 
     value: `numbers.Integral`
         A Python integral value (e.g. `int` or `numpy.integer`)
 
     storage_type: `str`, optional
-        Optionally overrides the storage type of the array used to represent
-        the integer value. Valid values are "internal" (the default) and
-        "inline"
-
-    Examples
-    --------
-
-    >>> import asdf
-    >>> import random
-    >>> # Create a large integer value
-    >>> largeval = random.getrandbits(100)
-    >>> # Store the large integer value to the tree using asdf.IntegerType
-    >>> tree = dict(largeval=asdf.IntegerType(largeval))
-    >>> with asdf.AsdfFile(tree) as af:
-    ...     af.write_to('largeval.asdf')
-    >>> with asdf.open('largeval.asdf') as aa:
-    ...     assert aa['largeval'] == largeval
+        This argument is now ignored.  Use asdf.get_config().array_inline_threshold
+        to customize the array storage type.
     """
 
-    name = "core/integer"
-    version = "1.0.0"
-    supported_versions = {"1.0.0", "1.1.0"}
-
-    _value_cache = {}
+    def __init__(self, value, storage_type=None):
+        warnings.warn(
+            "IntegerType is deprecated.  Large integer values are now handled automatically "
+            "and need no special wrapper class.  Use asdf.get_config().array_inline_threshold "
+            "to customize the array storage type.",
+            AsdfDeprecationWarning,
+        )
+
+        if storage_type is not None:
+            warnings.warn(
+                "The storage_type argument to IntegerType is now ignored.  Use "
```

Comment:
Would it be possible to retain this feature with the changes you added to the deferred block prototype (https://github.com/braingram/asdf/commit/90ddacc0a51df30bd348c99cb5597e72a14a58dd)?
I have not yet opened a PR to bring over those changes because I was going to include them with the 'adding block storage to the converter' PR that depends on #1468 #1469 #1473. If it would allow for this feature than perhaps we should create a PR for that feature (adding storage/compression settings for non-ndarray objects) first.